File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution 
{  
    public int findMinArrowShots(int[][] points) 
    {
        Arrays.sort(points, new Comparator<int[]>() {
                            public int compare(int[] x, int[] y) 
                            {
                                return (x[1] > y[1] ? 1 : -1);
                            }
                        });
        int val = points[0][1], result=1;
        for(int i=0; i<points.length; i++)
        {
            if(val < points[i][0])
            {
                result++;
                val = points[i][1];
            }
        }
        return result;
    }
}

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution 
{
    public:
    static bool compare(vector<int>& x, vector<int>& y)
    {   return x[1] < y[1];  }
    int findMinArrowShots(vector<vector<int>>& points) 
    {
        sort(points.begin(), points.end(), compare);
        int minx = points[0][1], result=1;
        for(int i=0; i<points.size(); i++)
            if(minx < points[i][0])
            {
                result++;
                minx = points[i][1];
            }
        return result;
    }
};